### ISSUE TITLE
http.server.requests error tag

### ISSUE THREAD

**Issue by chicobento** [spring-projects/spring-boot#33611]
On Spring Boot 3 besides `exception` , `method`, `outcome`, `status` and `uri`, the `error` tag is also being generated:

`http_server_requests_count{error="none" ... }`

This tag is [not on official docs](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#actuator.metrics.supported.spring-mvc) neither [ServerHttpObservationDocumentation](https://github.com/spring-projects/spring-framework/blob/v6.0.2/spring-web/src/main/java/org/springframework/http/server/observation/ServerHttpObservationDocumentation.java) class and overlaps with _exception_ tag. 
Looking at the code it seems that this behavior is added by micrometer's [DefaultMeterObservationHandler](https://github.com/micrometer-metrics/micrometer/blob/22a81f0cf1ac3f21e3b0b1f93a2b8c5c079422e2/micrometer-core/src/main/java/io/micrometer/core/instrument/observation/DefaultMeterObservationHandler.java#L74) which is configured by [ObservationAutoConfiguration](https://github.com/spring-projects/spring-boot/blob/c9c359fd271d1b51bf16f50d685c49fa40ca1b8a/spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/observation/ObservationAutoConfiguration.java#L123).

A possible quick-fix would be to override `ObservationAutoConfiguration#createErrorTags` to return `Tags.empty()`. 

On a side note, an additional undocumented behavior added by DefaultMeterObservationHandler is the `http.server.requests.active` metrics. Should this be documented ?

**Comment by bclozel** [comment]
I agree with @jonatan-ivanov 's assessment in this Slack thread.

I see 3 solutions for this:
1. Removing the "extra" `exception` tag contributied by Spring MVC. This would break a lot of existing applications and dashboards.
2. Removing the `error` tag contributed by Micrometer itself, since we're using the Micrometer Observation API. This doesn't make any sense as Micrometer can be used without Spring and should enforce such common tags
3. Leave things as is and live with the fact that we have duplicate tags.

I think 3) is the best solution. We could revisit this in the future, but I can't think of a proper milestone for this change. Typically, metrics dashboards deal with multiple applications, built with various Spring Boot versions. Unless there's a strong driver for this removal, I don't think this is worth breaking people.

I'm going to turn this into a documentation improvement and call out in our reference documentation that the `error` tag should be preferred in dashboards and that `exception` might be removed in the future.

**Comment by chicobento** [comment]
@bclozel fair enough, sounds like a plan. just a heads up that `http.server.requests.active` metric will have the exception tag but not the error tag - not that either one or the other are relevant for this metric. 
not sure about the behavior of other metrics though, will add here if I find anything.

**Comment by jonatan-ivanov** [comment]
@bclozel 👍🏼 Maybe it makes sense to provide the way in the docs to remove the exception tag if users want to:
```
@Bean
MeterFilter exceptionTagRemover() {
    return MeterFilter.ignoreTags("exception");
} 
```

A step forward could be (if users asks for it?) something similar that we have with common tags: if `management.metrics.tags` present, Boot creates a `MeterFilter` that adds those tags to every meter. We could do something very similar to ignore tags.
